### DISCUSSION TITLE
Need Help

### DISCUSSION THREAD

**Discussion by cheme75** [mrlt8/docker-wyze-bridge#328] [Q&A]
Really sorry for jumping in here since it appears I may be over my head, but I am a little frustrated - can’t find any info on how to even get started to set this up on a win10 machine - I have docker running just fine for another container, desktop shows it only using 1-5% cpu so thought I could add this to give it a try since I got a few Wyze cams I’d love to get into HomeKit. I downloaded the zip package, but clueless as to how to install it and get it setup in docker - appreciate if someone could point me in right direction.

**Comment by mrlt8** [comment]
Sorry about that. The dev branch (which is a preview of the next version) has a slightly updated readme with a [quick start guide](https://github.com/mrlt8/docker-wyze-bridge/tree/dev#quick-start) and updated [docker-compose](https://github.com/mrlt8/docker-wyze-bridge/tree/dev#docker-compose-recommended).

You don't actually need to download anything since the modern versions of the bridge have pre-built images.
You basically just need to use `docker run` with your wyze credentials and docker will download all the required images:
```
docker run \
  -e WYZE_EMAIL=you@email.com \
  -e WYZE_PASSWORD=yourpassw0rd \
  -p 8888:8888 -p 8554:8554 mrlt8/wyze-bridge:latest
```

You can view your stream by visiting: `http://localhost:8888/cam-nickname` where localhost is the hostname or ip of the machine running the bridge followed by the cam nickname in lowercase with - in place of spaces.

**Reply by cheme75** [reply]
Pasted above into txt file with my credentials, renamed to .cmd, then run as administrator - cmd window flashes with no errors but nothing happens - not seeing any download activity, nothing appears to be happening in docker.

**Reply by tbrausch** [reply]
> Sorry about that. The dev branch (which is a preview of the next version) has a slightly updated readme with a [quick start guide](https://github.com/mrlt8/docker-wyze-bridge/tree/dev#quick-start) and updated [docker-compose](https://github.com/mrlt8/docker-wyze-bridge/tree/dev#docker-compose-recommended).
> 
> You don't actually need to download anything since the modern versions of the bridge have pre-built images. You basically just need to use `docker run` with your wyze credentials and docker will download all the required images:
> 
> ```
> docker run \
>   -e WYZE_EMAIL=you@email.com \
>   -e WYZE_PASSWORD=yourpassw0rd \
>   -p 8888:8888 -p 8554:8554 mrlt8/wyze-bridge:latest
> ```
> 
> You can view your stream by visiting: `http://localhost:8888/cam-nickname` where localhost is the hostname or ip of the machine running the bridge followed by the cam nickname in lowercase with - in place of spaces.

Hoping someone can help me with, what I think, is a firewall issue.  First of all I am currently running Docker Wyze Bridge as a Home Assistant Addon and everything works perfectly.  However, the HA system is a little underpowered, so when I read this article I decided to try running Docker Wyze Bridge on my main windows PC.  I was easily able to install and get it running with the above instructions, and I can view HLS and RTSP streams on my windows PC.  However, when I change my lovelace config and frigate config on the HA system to point to the new IP address it doesn't work.  It also does not work from a chromebook on the local network.  

I thought that the firewall on Windows by default allowed communications on the local network, so is this a firewall issue or do I need to configure the docker container to allow access to other systems on the same subnet.  Any help would be appreciated.

**Reply by cheme75** [reply]
I did have to add firewall rules to allow port 8888 and 8554 for tcp on the private network on the win 10 pc running docker. I’m not using the other streams at present so didn’t bother with those yet.

**Reply by mrlt8** [reply]
Glad you were able to get it working.

You could also try [docker-compose](https://github.com/mrlt8/docker-wyze-bridge#docker-compose-recommended), so you can start it as a service and have run it in the background.

**Reply by cheme75** [reply]
> Glad you were able to get it working.
> 
> You could also try [docker-compose](https://github.com/mrlt8/docker-wyze-bridge#docker-compose-recommended), so you can start it as a service and have run it in the background.

just curious what -d detached mode is, don't see a reference?
I pretty sure the com.docker.service runs automatically on startup, but I think I have to allow the docker desktop to startup - been a while since I rebooted that machine so not exactly sure.

**Reply by mrlt8** [reply]
It just runs it in the [background](https://docs.docker.com/compose/reference/up/) so you don't need to keep a terminal window open.

**Reply by cheme75** [reply]
ok, used -d on last restart after pull for the latest update. Guess that's why I had to restart container from the docker desktop after I closed the terminal window before I used the -d, didn't dawn on me! once again, I appreciate the explanations - I keep bookmarking all these pages for future reference so hopefully I'll be more able to figure out stuff on my own.

**Comment by cheme75** [comment]
holy crap!!!! I got it working and its soooo cool!!! I used the docker-compose.yml sample, added my credentials, put it in bin folder under docker/docker/resounces, ran the docker-compose up, and bingo, I can view the streams in my browser - although they seem static until I press the play arrow. OMG this is so freaking amazing - no rtsp fw to fuss with is the best part. guess I'm not totally clueless for 70 yr old. now I just need to find out how to rotate the doorbell cams!

**Reply by mrlt8** [reply]
Awesome! 
Unfortunately, wyze encodes the doorbell streams in landscape mode, so the only way to rotate would be to re-encode the stream which will use more system resources. You can do this by editing your docker-compose.yml and adding `- ROTATE_DOOR=True` under `environment`, so something like this:

```yaml

        environment:
            - WYZE_EMAIL=${WYZE_EMAIL} # Replace with wyze email 
            - WYZE_PASSWORD=${WYZE_PASSWORD} # Replace with wyze password
            - ROTATE_DOOR=True

```

**Reply by cheme75** [reply]
Ok, will try that. I assume "DOOR" should be the actual device name like"front-door". 

Think I saw someone on Reddit mention a rotation of the rtsp stream. I have a couple of the beta doorbell cams setup just for live view streams since they power up using a usb wall charger and micro usb cable, and this picks them up as well as my actual doorbell and 2 other cams. If I can rotate it’s just a bit easier to view. Next is to see if my hoobs/homebridge server can see the docker pc so I can get the streams into homekit. I assume I'll need to open the ports on the docker machine. The docker machine doesn’t have WiFi, but it’s on the same subnet. So glad I found this - I really don't have any security needs but I love the challenge of getting tech stuff like this working.

**Reply by mrlt8** [reply]
`ROTATE_DOOR` is a special option that should rotate all the cams with the doorbell model number `WVOD1`, but you could use `- FFMPEG_CMD_CAM_NAME=ffmpeg...`If you wanted to pass your own ffmpeg commands to a specific camera.

Homebridge should be able to access the stream from the bridge as long as it's on the same LAN and you can access the stream on your host machine. Depending on your setup, you could also do the doorbell rotation in homebridge if that system can handle it better. See: https://sunoo.github.io/homebridge-camera-ffmpeg/configs/WyzeCam.html

Another option for homekit is [scrypted](https://github.com/mrlt8/docker-wyze-bridge/wiki/HomeKit-Secure-Video) which can do HKSV with object detection and cloud storage  or [homebridge-camera-ui](https://github.com/seydx/homebridge-camera-ui).

**Reply by cheme75** [reply]
Thx for details. Ill chk out the links. I guess I could set docker up on the old lubuntu machine for convenience. But it seems hoobs is about all it can handle, it’s an old 2010 era dual core pentium. The quad core i5 desktop is also old but seems to be running ok. I should be able to easily see each other on lan once I allow them to connect.

I’ll try the variable and see how it acts. Is there a list of all available variables by chance?

I assume I can change things in the yml file, then just do a restart of the container in docker to apply the updated yml?

I really appreciate the assistance.

**Reply by mrlt8** [reply]
Yup. `docker-compose up` or `docker-compose up -d` should reload the configuration and restart the container.

**Reply by cheme75** [reply]
> Yup. `docker-compose up` or `docker-compose up -d` should reload the configuration and restart the container.

added the - ROTATE_DOOR=True to the yml file, but get error when I run docker-compose up from cli, error says: /bin/sh: 1: docker-compose: not found when I try the win admin cmd prompt I get error: no configuration file provided: not found. would also like to be able to load the updated bridge, not clear on doing that - sorry for my lack of ability...

**Reply by mrlt8** [reply]
You probably need to `cd` to the directory where the yml file is located. You can enter `cd` then drag and drop the folder where the yml is to do so.

Another option would be to pass the location of the yml with something like:

```
docker-compose -f /path/to/my/docker-compose.yml up
```

As for updating the container, `cd` into the directory and run:

```
docker-compose pull # Pull new image
docker-compose up -d # Restart container in detached mode
docker image prune # Remove old images
```

**Reply by cheme75** [reply]
Yes sir! Thank you - got it restarted and rotate works perfectly - regarding the rotate variable, can it be applied to any cam - like -ROTATE_V2=True?  If not I'll eventually get it into homebridge and rotate there. I have one V2 on its side to get a doorbell type "portrait view" of my entry area, would be great to rotate it 90 deg.

All the early beta doorbells have the same model number as the final so they all seem to be recognized and seem to stream ok - although one of the betas has constant errors - if I recall correctly, this is why rev 3 failed - I'm gonna shut this one down: 
2022/04/27 15:24:05 [RTSP][SIDECAM] ❌ '/sidecam' stream is down
2022/04/27 15:24:27 [SideCam] [Errno 32] Broken pipe
2022/04/27 15:24:28 [WyzeBridge] 🎉 Connecting to WyzeCam Doorbell - SideCam on 192.168.1.161 (1/3)
2022/04/27 15:24:30 [SideCam] ☁️ WARNING: Camera is connected via RELAY mode. Stream may consume additional bandwidth!
2022/04/27 15:24:30 [SideCam] 📡 Getting 120kb/s HD stream @20fps via RELAY mode (WiFi: 88%) FW: 4.25.0.107 (2/3)
2022/04/27 15:24:30 [SideCam] WARNING: Skipping smaller frame at start of stream (frame_size=1)
2022/04/27 15:24:33 [SideCam] WARNING: Dropping old frames
2022/04/27 15:24:33 [SideCam] WARNING: Frame not available yet
2022/04/27 15:26:45 [RTSP][SIDECAM] ✅ '/sidecam' stream is UP! (3/3)

**Reply by mrlt8** [reply]
It only does the doorbell as the rotation can be quite resource intensive, so I'd try to avoid it if possible and do the rotation on the final playback device.

You can however pass your own custom commands to ffmpeg at any time using something like 
```yaml
- FFMPEG_CMD_FRONT_DOOR=ffmpeg -loglevel error -fflags +genpts+flush_packets+nobuffer -flags +low_delay -analyzeduration 50 -probesize 50 -f h264 -r 15 -i - -c:v libx264 -filter:v transpose=1 -b:v 3000K -tune zerolatency -preset ultrafast -force_key_frames expr:gte(t,n_forced*2) -movflags +empty_moov+default_base_moof+frag_keyframe -rtsp_transport tcp -f rtsp rtsp://0.0.0.0:8554/{cam_name}
```

**Reply by cheme75** [reply]
another problem - flushing out old devices - I replaced the beta sidecam with another beta cam on a new IP and restarted the wyze bridge, but it continues to try try to connect to the previous cam at the previous IP. How to I get rid of the old cam & IP ?

Also got this error on update attempt:
C:\Program Files\Docker\Docker\resources\bin>docker-compose pull
[+] Running 0/1
 - wyze-bridge Pulling                                                                                             0.3s
error getting credentials - err: docker-credential-desktop resolves to executable in current directory (.\docker-credential-desktop.exe), out: ``

**Reply by mrlt8** [reply]
You can set `- FRESH_DATA=true` to pull the latest camera list from the wyze api.

You could also set `- IGNORE_OFFLINE=true`

Or filter in/out the cams you want:

https://github.com/mrlt8/docker-wyze-bridge#examples

**Reply by cheme75** [reply]
What about this error?
I got this error on update attempt:
C:\Program Files\Docker\Docker\resources\bin>docker-compose pull
[+] Running 0/1

wyze-bridge Pulling 0.3s
error getting credentials - err: docker-credential-desktop resolves to executable in current directory (.\docker-credential-desktop.exe), out: ``
